Invisible units with advmodelshading

>[Kefirman] There's a problem after last update
[Kefirman] On some AMD cards units are invisible with enabled "shiny units" option
[Kefirman] Tested with 3 different PC
[Kefirman] 7660G have problem
[Kefirman] And 7750
[Kefirman] 7770* sry

https://pastebin.com/S9eyvX3C


```

[f=-000001] Custom Unit Shaders:, Fragment shader failed to compile with the following errors:
ERROR: 0:1: error(#76) Syntax error: unexpected tokens following #version
ERROR: 0:1: error(#364) Invalid: unexpected token in symbol.
ERROR: 0:4: error(#78) Syntax error: #else mismatch
ERROR: error(#273) 3 compilation errors.  No code generated
 
 
[f=-000001] Custom Unit Shaders:, Vertex shader failed to compile with the following errors:
ERROR: 0:31: error(#421) Keyword or operator "out" usage incorrect. GLSL version number 130 is required.
ERROR: 0:36: error(#421) Keyword or operator "out" usage incorrect. GLSL version number 130 is required.
ERROR: 0:39: error(#421) Keyword or operator "out" usage incorrect. GLSL version number 130 is required.
ERROR: 0:40: error(#421) Keyword or operator "out" usage incorrect. GLSL version number 130 is required.
ERROR: error(#273) 4 compilation errors.  No code generated
 
 
[f=-000001] Custom Unit Shaders:, Fragment shader failed to compile with the following errors:
ERROR: 0:6: error(#78) Syntax error: #else mismatch
ERROR: error(#273) 1 compilation errors.  No code generated
 
 
[f=-000001] Custom Unit Shaders:, Fragment shader failed to compile with the following errors:
ERROR: 0:6: error(#78) Syntax error: #else mismatch
ERROR: error(#273) 1 compilation errors.  No code generated
 
 
[f=-000001] Custom Unit Shaders:, Fragment shader failed to compile with the following errors:
ERROR: 0:5: error(#78) Syntax error: #else mismatch
ERROR: error(#273) 1 compilation errors.  No code generated
 
```

@lhog ?

I can confirm it does with every graphic cards, not just ATI and it affects every units and buildings with a reflective effect.
Got this in infolog:

```
[f=-000001] Custom Unit Shaders:, 0(1) : error C0206: invalid token "<invalid atom -136938312>" in version line
0(4) : error C7102: unmatched #else
0(6) : error C0122: #else cannot follow #else

[f=-000001] Custom Unit Shaders:, 0(3) : warning C7022: unrecognized profile specifier "highp"
0(3) : warning C7022: unrecognized profile specifier "precision"
0(52) : warning C5060: out can't be used with non-varying fragData
0(81) : warning C7532: global function texture requires "#version 130" or later
0(87) : warning C7532: global function texture requires "#version 130" or later

[f=-000001] Custom Unit Shaders:, 0(6) : error C7102: unmatched #else
0(8) : error C0122: #else cannot follow #else

[f=-000001] Custom Unit Shaders:, 0(6) : error C7102: unmatched #else
0(8) : error C0122: #else cannot follow #else

[f=-000001] Custom Unit Shaders:, 0(5) : error C7102: unmatched #else
0(7) : error C0122: #else cannot follow #else
```

So I think https://github.com/ZeroK-RTS/Zero-K/commit/5b15c005d9dcbbe78b506e8669da59187687d26f is at fault.

--- a/LuaRules/Gadgets/api_custom_unit_shaders.lua
+++ b/LuaRules/Gadgets/api_custom_unit_shaders.lua
@@ -147,10 +147,10 @@ local function CompileShader(shader, definitions, plugins)
     if not hasVersion then
       table.insert(definitions, 1, "#version 130")
     end
-    definitions = table.concat(definitions, "\n")
     if (shadows) then
-      definitions = definitions .. "\n" .. "#define use_shadows" .. "\n"
+      table.insert(definitions, "#define use_shadows")
     end
+    definitions = table.concat(definitions, "\n") .. "\n"
     if (shader.vertex)
       then shader.vertex = definitions .. shader.vertex; end
     if (shader.fragment)

